pkgs/by-name/st/steel/package.nix

```diff
@@ -0,0 +1,106 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  curl,
+  pkg-config,
+  makeBinaryWrapper,
+  libgit2,
+  oniguruma,
+  openssl,
+  sqlite,
+  zlib,
+  darwin,
+
+  includeLSP ? true,
+  includeForge ? true,
+}:
+rustPlatform.buildRustPackage {
+  pname = "steel";
+  version = "0.6.0-unstable-2025-02-08";
+
+  src = fetchFromGitHub {
+    owner = "mattwparas";
+    repo = "steel";
+    rev = "3b7b9812e02f08cf123d0b28ef675f3522f355a4";
+    hash = "sha256-htQ4phadl54oxISegKXTVAx26JEXz3hGv8saAd9CHQs=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-8S/CJFXv89lS3Gyd/TgHUrMxmvYQZvbXCiPQcQtL3Jo=";
+
+  nativeBuildInputs = [
+    curl
+    makeBinaryWrapper
+    pkg-config
+    rustPlatform.bindgenHook
+  ];
+
+  buildInputs = [
+    curl
+    libgit2
+    oniguruma
+    openssl
+    sqlite
+    zlib
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+
+  cargoBuildFlags =
+    [
+      "--package"
+      "steel-interpreter"
+      "--package"
+      "cargo-steel-lib"
+    ]
+    ++ lib.optionals includeLSP [
+      "--package"
+      "steel-language-server"
+    ]
+    ++ lib.optionals includeForge [
+      "--package"
+      "forge"
+    ];
+
+  # Tests are disabled since they always fail when building with Nix
+  doCheck = false;
+
+  postInstall = ''
+    mkdir -p $out/lib/steel
+
+    substituteInPlace cogs/installer/download.scm \
+      --replace-fail '"cargo-steel-lib"' '"$out/bin/cargo-steel-lib"'
+
+    pushd cogs
+    $out/bin/steel install.scm
+    popd
+
+    mv $out/lib/steel/bin/repl-connect $out/bin
+    rm -rf $out/lib/steel/bin
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/steel --set-default STEEL_HOME "$out/lib/steel"
+  '';
+
+  env = {
+    OPENSSL_NO_VENDOR = true;
+    RUSTONIG_SYSTEM_LIBONIG = true;
+    STEEL_HOME = "${placeholder "out"}/lib/steel";
+  };
+
+  passthru.updateScript = ./update.sh;
```

Comment:
may be able to use nixUnstableUpdater instead?

Reply:
are you talking about `unstableGitUpdater`? I probably can use it but I don't know if I want to, because I would have to write another script for the tagConverter arg, so I could add the version prefix in the version, because there isn't any tags.

Reply:
Yeah, I meant that. I dunno. It may be fine

Reply:
Ended up switching to `unstableGitUpdater` as my script didn't always work correctly.